File: alarmclock/src/test/java/com/externalalarmclock/alarmclock/ChannelFactoryTest.java
package com.externalalarmclock.alarmclock;

import org.junit.Assert;
import org.junit.Test;

import com.externalalarmclock.lib.rpiws281x.ERpiWs281xStripType;
import com.externalalarmclock.lib.rpiws281x.RpiWs281xChannel;

public class ChannelFactoryTest {
	@Test
	public void testChannelFactory() {
		ChannelFactory factory = new ChannelFactory();
		
		RpiWs281xChannel channel = factory.createChannel();
		
		Assert.assertEquals(-1, channel.getGpioNum());
		Assert.assertEquals(-1, channel.getLedCount());
		Assert.assertEquals(-1, channel.getBrightness());
		Assert.assertEquals(ERpiWs281xStripType.SK6812_STRIP_GRBW, channel.getStripType());
	}
}

File: alarmclock/src/test/kotlin/com/externalalarmclock/alarmclock/ChannelFactoryTest.kt
package com.externalalarmclock.alarmclock

import com.externalalarmclock.lib.rpiws281x.ERpiWs281xStripType
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class ChannelFactoryTest {
    @Test
    fun testChannelFactory() {
        val factory = ChannelFactory()

        val channel = factory.createChannel()

        Assertions.assertEquals(-1, channel.gpioNum.toLong())
        Assertions.assertEquals(-1, channel.ledCount.toLong())
        Assertions.assertEquals(-1, channel.brightness.toLong())
        Assertions.assertEquals(ERpiWs281xStripType.SK6812_STRIP_GRBW, channel.stripType)
    }
}
